### PR TITLE
implement strictPopulateError

### PR DESCRIPTION
**Pull request by IslandRhythms**

### CODE REVIEW

File: lib/error/strictPopulate.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+   * @param {String} path
+   * @param {String} [msg]
+   * @param {Boolean} [immutable]
+   * @inherits MongooseError
+   * @api private
+   */
+  constructor(path, msg, immutable) {
+    msg = msg || 'Cannot populate path `' + path + '` because it is not in your schema. ' + 'Set the `strictPopulate` option to false to override.';
+    super(msg);
+    this.isImmutableError = !!immutable;
```
Comment (vkarpov15) [inline lib/error/strictPopulate.js:23]: Please remove this line, not relevant to this error

File: lib/error/strictPopulate.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  /**
+   * Strict mode error constructor
+   *
+   * @param {String} path
+   * @param {String} [msg]
+   * @param {Boolean} [immutable]
+   * @inherits MongooseError
+   * @api private
+   */
+  constructor(path, msg, immutable) {
```
Comment (vkarpov15) [inline lib/error/strictPopulate.js:20]: Remove `immutable` in constructor

File: lib/error/strictPopulate.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+const MongooseError = require('./');
+
+
+class StrictPopulateError extends MongooseError {
+  /**
+   * Strict mode error constructor
+   *
+   * @param {String} path
+   * @param {String} [msg]
+   * @param {Boolean} [immutable]
```
Comment (vkarpov15) [inline lib/error/strictPopulate.js:16]: And remove immutable here please

**Review by vkarpov15 — COMMENTED** [review]
Couple of quick comments, otherwise looks good

File: lib/error/strictPopulate.js (hunk 1 above)
Comment (Uzlopak) [reply lib/error/strictPopulate.js:23]:
```suggestion
```

File: lib/error/strictPopulate.js (hunk 2 above)
Comment (Uzlopak) [reply lib/error/strictPopulate.js:20]:
```suggestion
  constructor(path, msg) {
```

File: lib/error/strictPopulate.js (hunk 3 above)
Comment (Uzlopak) [reply lib/error/strictPopulate.js:16]:
```suggestion
```

File: lib/error/strictPopulate.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+/*!
+ * Module dependencies.
+ */
+
+'use strict';
+
+const MongooseError = require('./');
+
+
```
Comment (Uzlopak) [inline lib/error/strictPopulate.js:9]:
```suggestion
```

**Pull request merged**